site/en/blog/framework-tools-font-fallback/index.md

```diff
@@ -0,0 +1,402 @@
+---
+layout: 'layouts/blog-post.njk'
+title: Framework tools for font fallbacks
+description: 
+  Learn how Next.js, Nuxt.js, and other libraries help to generate fallback font CSS without CLS.
+date: 2023-02-24
+authors:
+  - janicklasralphj
+hero: 'image/IypihH3o5cSpEMVp5i08dp69otp2/aBnMiacBdhhgGBozA5jJ.jpg'
+alt: >
+  The alphabet A in different sizes
+tags:
+  - performance
+  - aurora-project
+---
+
+Sites that load fonts with _font-display: swap_ often suffer from a layout shift ([CLS](https://web.dev/cls/)) when the web font loads and is swapped with the fallback font.
+
+You can prevent CLS by adjusting the dimensions of the fallback font to match that of the primary font. Properties such as size-adjust, ascent-override, decent-override, and line-gap-override in the [@font-face](https://developer.mozilla.org/docs/Web/CSS/@font-face) rule can help override the metrics of a fallback font allowing developers more control over how fonts are displayed. You can read more about font-fallbacks and the override properties in this [post](/blog/font-fallbacks/). You can also see a working implementation of this technique in this [demo](https://cultured-rightful-check.glitch.me).
+
+This article explores how font size adjustments are implemented in the Next.js and Nuxt.js frameworks to generate the fallback font CSS and reduce the CLS. It also demonstrates how you can generate fallback fonts using tools such as Fontaine and Capsize. 
+
+
+## Background
+
+_[font-display: swap](/blog/font-display/#swap)_ is generally used to prevent FOIT (Flash of invisible text) and to display contents faster on the screen. The value of _swap_ tells the browser that text using the font should be displayed immediately using a system font and to replace the system font only when the custom font is ready.
```

Comment:
```suggestion
_[font-display: swap](/blog/font-display/#swap)_ is generally used to prevent FOIT (Flash of invisible text) and to display contents faster on the screen. The value of `swap` tells the browser that text using the font should be displayed immediately using a system font and to replace the system font only when the custom font is ready.
```